pkg/operator/controller/ingress/internal_service_test.go

```diff
@@ -116,6 +116,22 @@ func Test_internalServiceChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if .spec.ports changes by changing the metrics port's target port from an integer to a string",
+			mutate: func(svc *corev1.Service) {
+				foundPort := false
+				for i := range svc.Spec.Ports {
+					if svc.Spec.Ports[i].Name == "metrics" {
+						svc.Spec.Ports[i].TargetPort = intstr.FromString("metrics")
+						foundPort = true
```

Comment:
Can we return early here? if so, then we don't need `foundPort`. If there's no match we can just panic.